### PR TITLE
refactor: recommended modal refactor

### PR DESCRIPTION
**Pull request by nada-deriv**
## Changes:

refactor recommended modal

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

**Review by ameerul-deriv — APPROVED** [review]
LGTM

File: packages/p2p/src/components/modal-manager/modals/recommended-modal/recommended-modal.tsx [hunk 1]
```diff
@@ -15,14 +18,10 @@ const RecommendedModal = ({ message }) => {
                 </Text>
             </Modal.Body>
             <Modal.Footer>
-                <Button large primary onClick={hideModal} text={localize('Ok')} />
+                <Button large primary onClick={() => hideModal()} text={localize('Ok')} />
```
Comment (ako-deriv) [inline packages/p2p/src/components/modal-manager/modals/recommended-modal/recommended-modal.tsx:21]: whats the reason behind this?

Comment (ameerul-deriv) [reply packages/p2p/src/components/modal-manager/modals/recommended-modal/recommended-modal.tsx:21]: we get this eslint issue if we don't do it this way when passing hideModal to onClick

this is the error
`Type '(options?: THideModalOptions | undefined) => void' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
  Types of parameters 'options' and 'event' are incompatible.
    Type 'MouseEvent<HTMLButtonElement, MouseEvent>' has no properties in common with type 'THideModalOptions'.`

File: packages/p2p/src/components/modal-manager/modals/recommended-modal/__tests__/recommended-modal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import { useStores } from 'Stores';
+import RecommendedModal from '../recommended-modal';
+
+const mocked_store_values: DeepPartial<ReturnType<typeof useStores>> = {
+    my_profile_store: {
+        add_payment_method_error_message: 'this is the error message',
+        setAddPaymentMethodErrorMessage: jest.fn(),
```
Comment (farrah-deriv) [inline packages/p2p/src/components/modal-manager/modals/recommended-modal/__tests__/recommended-modal.spec.tsx]: why do we need to mock these?

Comment (nada-deriv) [reply packages/p2p/src/components/modal-manager/modals/recommended-modal/__tests__/recommended-modal.spec.tsx]: removed.

**Review by adrienne-deriv — APPROVED** [review]
LGTM

**Review by farhan-nurzi-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]

**Pull request merged**